### PR TITLE
Fix incorrect filtering on column name with dot in parquet

### PR DESCRIPTION
**Pull request by raunaqmorarka**

### CODE REVIEW

**Comment by findepi** [comment]
cc @alexjo2144 @jirassimok @losipiuk

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkipping.java [hunk 1]
```diff
@@ -118,6 +118,23 @@ public void testPageSkippingWithNonSequentialOffsets()
...
+        String columnName = "a.dot";
+        String nameInSql = "\"" + columnName.replace("\"", "\"\"") + "\"";
+
+        String tableName = "test_column_name_with_dot_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + "(key varchar(50), " + nameInSql + " varchar(50)) WITH (format = 'PARQUET')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('null value', NULL), ('sample value', 'abc'), ('other value', 'xyz')", 3);
+
+        assertQuery("SELECT key FROM " + tableName + " WHERE " + nameInSql + " IS NULL", "VALUES ('null value')");
+        assertQuery("SELECT key FROM " + tableName + " WHERE " + nameInSql + " = 'abc'", "VALUES ('sample value')");
```
[the remark this answers is not in the document]
Comment (raunaqmorarka) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkipping.java:133]: Indeed, that's how I stumbled onto this issue

File: lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java [hunk 2]
```diff
@@ -604,4 +607,17 @@ private DictionaryValueConverter(Dictionary dictionary)
             }
         }
     }
+
+    // FilterApi#intColumn splits column name on ".". If column name contains a "." this leads to
+    // ColumnIndexFilter#calculateRowRanges failing to detect that column as part of the projection
```
[the remark this answers is not in the document]
Comment (raunaqmorarka) [reply lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java:612]: I couldn't find anything about handling this in parquet-mr

[the remark this answers is not in the document]
Comment (jirassimok) [reply lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java:612]: From the discussion on [this issue](https://issues.apache.org/jira/browse/PARQUET-1809) ("Add new APIs for nested predicate pushdown"), it looks like there is no built-in way to handle this.

**Pull request merged**